lib/events.coffee

```diff
@@ -30,3 +30,5 @@ exports.trackCommand = (capitanoCommand) ->
 				resinUrl: resinUrl
 				platform: process.platform
 				command: capitanoCommand
+	.timeout(100)
```

Comment:
Lets say that there is a command runs in 1ms and the tracking code for some reason needs 150ms (eg slow DNS or overcrowded wifi).
Is there a chance that the cli closes and everything gets disposed before the mixpanel tracking completes and we loose that stats?
Would it make any difference to increase the timeout to something a bit larger?
(I'm not sure if that makes sense, but I was thinking of it last night, for cli commands that can complete without any network requests.)

Reply:
> Would it make any difference to increase the timeout to something a bit larger?

Maybe. I've gone for a 100ms to avoid this issue for cases even when it's super fast, but in slow conditions that can still happen. It's not too bad though: I think as long as the full request has been sent, it doesn't matter if the response hasn't come back, so events should be tracked even if we time out while waiting for the mixpanel server & response. If we time out in TCP connection setup on the other hand, yes, we're stuck.

This problem is always going to be present up to a point though - eventually we have to decide whether we want to make users wait, or lose events, and I think I'm fairly happy dropping events occasionally in those cases. We could boost the timeout... I was trying to pick a time that's going to generally be sufficient, but isn't going to ever be noticeable if users do have to wait for it.

For now I've bumped it to a 500ms max. That could still time out, but it feels like it should successfully fire the event first, and if we block this beyond that point then we're well into noticeable experience, and even if we're going to lose events, I don't want to make users wait.

Reply:
:+1: